packages/dashboard/src/util/resizeSelectionBox.ts

```diff
@@ -0,0 +1,42 @@
+import type { Position, Rect } from '~/types';
+import type { Anchor } from '~/store/actions';
+
+const MIN_WIDTH = 2;
+const MIN_HEIGHT = 2;
+const rectWithinMin = (rect: Rect): Rect => {
+  const { width, height } = rect;
+
+  return {
+    ...rect,
+    width: Math.max(width, MIN_WIDTH),
+    height: Math.max(height, MIN_HEIGHT),
+  };
+};
+export const resizeSelectionBox: (params: { selectionBox: Rect; anchor: Anchor; vector: Position }) => Rect = ({
+  selectionBox,
+  anchor,
+  vector,
+}) => {
+  const newRect = { ...selectionBox };
+  if (anchor.includes('top')) {
+    if (newRect.height - vector.y >= MIN_HEIGHT) {
+      newRect.y += vector.y;
+      newRect.height -= vector.y;
+    }
+  }
+  if (anchor.includes('left')) {
+    // don't allow the selection box to be smaller than the minimum width
+    if (newRect.width - vector.x >= MIN_WIDTH) {
+      newRect.x += vector.x;
+      newRect.width -= vector.x;
+    }
+  }
+  if (anchor.includes('right')) {
+    newRect.width += vector.x;
```

Comment:
should we do a min width check here too? The vector can be negative

Reply:
width/height checking is at end of the function by `rectWithinMin`

Reply:
We only need to check if x/y are moved here

Reply:
so the specific checks on left and top are there to prevent the x value from pushing?

Reply:
Yes. So if widget doesn't get resized on width or height, don't move the origin point (x,y).

Reply:
we only need to check moving from top or left, since only these two directions modify x/y.

Reply:
gotcha, perfect!